python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -0,0 +1,260 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy
+
+import lsstDebug
+import lsst.afw.display.ds9 as ds9
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+__all__ = ("SubtractBackgroundConfig", "SubtractBackgroundTask")
+
+class SubtractBackgroundConfig(pexConfig.Config):
+    """!Config for SubtractBackgroundTask and lsst::afw::math::makeBackground
+
+    @warning This config is automatically converted to an lsst::afw::math::BackgroundControl
+    for lsst::afw::math::makeBackground, so do not change the field names or types.
+    """
+    statisticsProperty = pexConfig.ChoiceField(
+        doc="type of statistic to use for grid points",
+        dtype=str, default="MEANCLIP",
+        allowed={
+            "MEANCLIP": "clipped mean",
+            "MEAN": "unclipped mean",
+            "MEDIAN": "median",
+            }
+    )
+    undersampleStyle = pexConfig.ChoiceField(
+        doc="behaviour if there are too few points in grid for requested interpolation style",
+        dtype=str, default="REDUCE_INTERP_ORDER",
+        allowed={
+            "THROW_EXCEPTION": "throw an exception if there are too few points",
+            "REDUCE_INTERP_ORDER": "use an interpolation style with a lower order.",
+            "INCREASE_NXNYSAMPLE": "Increase the number of samples used to make the interpolation grid.",
+            },
+    )
+    binSize = pexConfig.RangeField(
+        doc="how large a region of the sky should be used for each background point",
+        dtype=int, default=256, min=10,
+    )
+    algorithm = pexConfig.ChoiceField(
+        doc="how to interpolate the background values. This maps to an enum; see afw::math::Background",
+        dtype=str, default="NATURAL_SPLINE", optional=True,
+        allowed={
+            "CONSTANT" : "Use a single constant value",
+            "LINEAR" : "Use linear interpolation",
+            "NATURAL_SPLINE" : "cubic spline with zero second derivative at endpoints",
+            "AKIMA_SPLINE": "higher-level nonlinear spline that is more robust to outliers",
+            "NONE": "No background estimation is to be attempted",
+            },
+    )
+    ignoredPixelMask = pexConfig.ListField(
+        doc="Names of mask planes to ignore while estimating the background",
+        dtype=str, default = ["BAD", "EDGE", "DETECTED", "DETECTED_NEGATIVE", "NO_DATA",],
+        itemCheck = lambda x: x in afwImage.MaskU().getMaskPlaneDict().keys(),
+    )
+    isNanSafe = pexConfig.Field(
+        doc="Ignore NaNs when estimating the background",
+        dtype=bool, default=False,
```

Comment:
I think that is out of scope for this ticket, which consists of refactoring, not changing behavior.